Userland/Libraries/LibGfx/VectorN.h

```diff
@@ -0,0 +1,263 @@
+#    pragma unroll 4
+#else
+#    pragma GCC unroll 4
+#endif
+        for (auto i = 0u; i < N; ++i) {
+            m_data[i] += other.m_data[i];
+        }
+        return *this;
+    }
+
+    constexpr VectorN& operator-=(const VectorN& other)
+    {
+#ifdef __clang__
+#    pragma unroll 4
+#else
+#    pragma GCC unroll 4
+#endif
+        for (auto i = 0u; i < N; ++i) {
+            m_data[i] -= other.m_data[i];
+        }
+        return *this;
+    }
+
+    constexpr VectorN operator+(const VectorN& other) const
+    {
+        VectorN result {};
+#ifdef __clang__
+#    pragma unroll 4
+#else
+#    pragma GCC unroll 4
+#endif
+        for (auto i = 0u; i < N; ++i) {
+            result.m_data[i] = m_data[i] + other.m_data[i];
+        }
+        return result;
+    }
+
+    constexpr VectorN operator-(const VectorN& other) const
+    {
+        VectorN result {};
+#ifdef __clang__
+#    pragma unroll 4
+#else
+#    pragma GCC unroll 4
+#endif
+        for (auto i = 0u; i < N; ++i) {
+            result.m_data[i] = m_data[i] - other.m_data[i];
+        }
+        return result;
+    }
+
+    constexpr VectorN operator*(const VectorN& other) const
+    {
+        VectorN result {};
+#ifdef __clang__
+#    pragma unroll 4
+#else
+#    pragma GCC unroll 4
+#endif
+        for (auto i = 0u; i < N; ++i) {
+            result.m_data[i] = m_data[i] * other.m_data[i];
+        }
+        return result;
+    }
+
+    constexpr VectorN operator-() const
+    {
+        VectorN result {};
+#ifdef __clang__
+#    pragma unroll 4
+#else
+#    pragma GCC unroll 4
+#endif
+        for (auto i = 0u; i < N; ++i) {
+          result.m_data[i] = -m_data[i];
+        }
+        return result;
+    }
+
+    constexpr VectorN operator/(const VectorN& other) const
+    {
+        VectorN result {};
+#ifdef __clang__
+#    pragma unroll 4
+#else
+#    pragma GCC unroll 4
+#endif
+        for (auto i = 0u; i < N; ++i) {
+            result.m_data[i] = m_data[i] / other.m_data[i];
+        }
+        return result;
+    }
+
+    template<typename U>
+    constexpr VectorN operator*(U f) const
+    {
+        VectorN result {};
+#ifdef __clang__
+#    pragma unroll 4
+#else
+#    pragma GCC unroll 4
+#endif
+        for (auto i = 0u; i < N; ++i) {
+            result.m_data[i] = m_data[i] * f;
+        }
+        return result;
+    }
+
+    template<typename U>
+    constexpr VectorN operator/(U f) const
+    {
+        VectorN result {};
+#ifdef __clang__
+#    pragma unroll 4
+#else
+#    pragma GCC unroll 4
+#endif
+        for (auto i = 0u; i < N; ++i) {
+            result.m_data[i] = m_data[i] / f;
+        }
+        return result;
+    }
+
+    constexpr T dot(const VectorN& other) const
+    {
+        T result {};
+        result += x() * other.x();
+        result += y() * other.y();
+        if constexpr (N >= 3) {
+            result += z() * other.z();
+        }
+        if constexpr (N >= 4) {
+            result += w() * other.w();
+        }
```

Comment:
Why not loop over `m_data` here as well?

Reply:
Good call. I missed it.